source/updateCheck.py

```diff
@@ -94,6 +97,66 @@
 autoChecker: Optional["AutoUpdateChecker"] = None
 
 
+@dataclass
+class UpdateInfo:
+	"""
+	Data class representing update information for NVDA.
+
+	Attributes:
+		version (str): The version of the update.
+		launcherUrl (str): The URL to download the launcher.
+		apiVersion (str): The version of the API.
+		launcherHash (Optional[str]): The hash of the launcher, if available.
+		apiCompatTo (Optional[str]): The API compatibility version, if available.
+		changesUrl (Optional[str]): The URL to the changelog, if available.
+		launcherInteractiveUrl (Optional[str]): The URL for the interactive launcher, if available.
+	"""
+
+	version: str
+	launcherUrl: str
+	apiVersion: str
+	launcherHash: Optional[str] = None
+	apiCompatTo: Optional[str] = None
+	changesUrl: Optional[str] = None
+	launcherInteractiveUrl: Optional[str] = None
+
+	@classmethod
+	def parseUpdateCheckResponse(cls, data: str) -> Optional["UpdateInfo"]:
+		"""
+		Parses the update response and returns an UpdateInfo object.
+
+		:param data: The raw server response as a UTF-8 decoded string.
+		:return: An UpdateInfo object containing the update metadata, or None if the format is invalid.
+		:raises ValueError: If the response format is invalid.
+		"""
+		required_keys = ["version", "launcherUrl", "apiVersion"]
+		metadata = {}
+		for line in data.splitlines():
+			try:
+				key, val = line.split(": ", 1)
+				metadata[key] = val
+			except ValueError:
+				raise ValueError(f"Invalid line format in update response: {line}")
+
+		# Check for missing required keys
+		missing_keys = [key for key in required_keys if key not in metadata]
+		if missing_keys:
+			raise ValueError(f"Missing required keys in update response: {', '.join(missing_keys)}")
+
+		# Create and return UpdateInfo
+		return cls(
+			# Required keys
+			version=metadata["version"],
+			launcherUrl=metadata["launcherUrl"],
+			apiVersion=metadata["apiVersion"],
+			# Optional keys
+			launcherHash=metadata.get("launcherHash", None),
+			apiCompatTo=metadata.get("apiCompatTo", None),
+			changesUrl=metadata.get("changesUrl", None),
+			launcherInteractiveUrl=metadata.get("launcherInteractiveUrl", None),
+		)
```

Comment:
It would probably be easier to do
```suggestion
		return cls(**metadata)
```
The further suggestions above should help to make this fairly bullet-proof,, but if you wanted to you could wrap this in a try/except, except `TypeError`, and re-raise it as a `ValueError` since that's more appropriate here.